### PR TITLE
Dom implementation

### PR DESCRIPTION
**Pull request by alexcope**
Added the ability to navigate the DOM, Highlight nodes, and added the ability to execute commands in the console

### CODE REVIEW

File: IEWebkitImpl/Browser.ts [hunk 1]
```diff
@@ -161,7 +163,9 @@ module F12.Proxy {
                         case "Page":
                             this.ProcessPage(methodParts[1], request);
                             break;
-
+                        case "DOM":
+                            domHandler.ProcessDOM(methodParts[1], request);
```
Comment (jalissia) [inline IEWebkitImpl/Browser.ts]: Does domHandler need to be global? Couldn't it just be constructed as a member to this class?

File: IEWebKitImpl/DOM.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                margin: "rgba(250, 212, 107, 0.50)",
+                border: "rgba(120, 181, 51, 0.50)",
+                padding: "rgba(247, 163, 135, 0.50)",
+                content: "rgba(168, 221, 246, 0.50)"
+            };
+        }
+
+        private createChromeNodeFromIENode(node: Node): INode {
+            var inode: INode = {
+                nodeId: +this.getOrAssignUid(node),
```
Comment (jalissia) [inline IEWebKitImpl/DOM.ts]: nodeId: +this.getOrAssignUid(node),
What is that + for?

File: IEWebKitImpl/DOM.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                    inode.attributes.push(node.attributes[i].value);
+                }
+            }
+
+            return inode;
+        }
+
+        public getOrAssignUid(node: Node): number {
+            if (!node) {
+                return;
```
Comment (jalissia) [inline IEWebKitImpl/DOM.ts]: void isn't really a number

File: IEWebKitImpl/DOM.ts [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+            this._elementHighlightColor = {
+                margin: "rgba(250, 212, 107, 0.50)",
+                border: "rgba(120, 181, 51, 0.50)",
+                padding: "rgba(247, 163, 135, 0.50)",
+                content: "rgba(168, 221, 246, 0.50)"
+            };
+        }
+
+        private createChromeNodeFromIENode(node: Node): INode {
+            var inode: INode = {
```
Comment (jalissia) [inline IEWebKitImpl/DOM.ts]: iNode

File: IEWebKitImpl/DOM.ts (hunk 2 above)
Comment (alexcope) [inline IEWebKitImpl/DOM.ts]: Originally I used it to convert strings to numbers, it is no longer needed so I'll remove it

**Pull request merged**